components/HomepageSection/Activity.jsx

```diff
@@ -1,38 +1,67 @@
 import { useMemo } from 'react';
 import Link from 'next/link';
 import Image from 'next/image';
-import PropTypes from 'prop-types';
-
+import useSWR from 'swr';
+import {
+  getTotalTransactionsCount,
+  getTotalUnitsCount,
+  get7DaysAvgActivity,
+} from 'common-util/api/flipside';
 import SectionWrapper from 'components/Layout/SectionWrapper';
-// import { Button } from 'components/ui/button';
 import { ExternalLink } from 'components/ui/typography';
 import { Popover } from 'components/ui/popover';
-import { FLIPSIDE_URL, DUNE_QUERY_URL, DAILY_ACTIVE_AGENTS_DUNE_QUERY_ID } from 'common-util/constants';
+import { FLIPSIDE_URL } from 'common-util/constants';
 import { Card } from 'components/ui/card';
 import SectionHeading from '../SectionHeading';
 
 const BLOCKCHAIN_COUNT = 8;
 
-export const Activity = ({
-  activityMetrics: {
-    agents, agentsTypes, transactions, dailyActiveAgents,
-  },
-}) => {
+const fetchMetrics = async () => {
+  const [transactions, unitsCount, dailyActiveAgents] = await Promise.allSettled([
+    getTotalTransactionsCount(),
+    getTotalUnitsCount(),
+    get7DaysAvgActivity(),
+  ]);
+
+  return {
+    transactions:
+      transactions.status === 'fulfilled' ? transactions.value : null,
+    agents:
+      unitsCount.status === 'fulfilled'
+        ? unitsCount.value.agentsCount
+        : null,
+    agentsTypes:
+      unitsCount.status === 'fulfilled'
+        ? unitsCount.value.agentTypesCount
+        : null,
+    dailyActiveAgents:
+      dailyActiveAgents.status === 'fulfilled' ? dailyActiveAgents.value : null,
+  };
+};
+
+const usePersistentSWR = (key, fetcher) => useSWR(key, fetcher, {
+  revalidateIfStale: false,
+  revalidateOnFocus: false,
+});
+
+export const Activity = () => {
+  const { data: metrics } = usePersistentSWR('activityMetrics', fetchMetrics);
```

Comment:
use swr here so the data is cached and not re-requested when we visit another page and go back